ql/src/java/org/apache/hadoop/hive/ql/exec/repl/ReplDumpTask.java

```diff
@@ -180,6 +184,21 @@ public int execute() {
         }
         //If no previous dump is present or previous dump is already loaded, proceed with the dump operation.
         if (shouldDump(previousValidHiveDumpPath)) {
+          if (previousValidHiveDumpPath != null) {
+            FileSystem fs = previousValidHiveDumpPath.getFileSystem(conf);
+            if (fs.exists(new Path(previousValidHiveDumpPath, ReplAck.FAILOVER_READY_MARKER.toString()))) {
+              Database db = getHive().getDatabase(work.dbNameOrPattern);
+              if (MetaStoreUtils.isDbBeingFailedOver(db)) {
+                unsetReplFailoverEnabled(db);
+                fs.delete(new Path(previousValidHiveDumpPath, ReplAck.FAILOVER_READY_MARKER.toString()));
+              } else {
+                LOG.info("Switching to bootstrap dump as this is the first dump execution after failover.");
+                isBootstrap = true;
```

Comment:
We don't need to use a separate variable isBootstrap given that work already has it